### PR TITLE
Validate privileged mode with edge-case tests and documentation

### PR DESCRIPTION
**Pull request by leynos**
Add dedicated unit tests and BDD scenarios confirming that privileged mode (sandbox.privileged = true) intentionally ignores mount_dev_fuse and SELinux label mode settings, because the engine host profile governs device access and security labelling directly.

Unit tests added:
- privileged mode ignores mount_dev_fuse = false
- privileged mode ignores SELinux DisableForContainer override
- from_sandbox_config non-privileged derives DisableForContainer
- privileged mode without optional fields (name, cmd, env)

BDD scenarios added:
- Privileged mode ignores /dev/fuse setting
- Privileged mode ignores SELinux override

Documentation updated:
- podbot-design.md: note that FUSE and SELinux settings are ignored in privileged mode
- users-guide.md: note that mount_dev_fuse has no effect in privileged mode
- podbot-roadmap.md: mark task 2.2.2 as done

Execution plan written to docs/execplans/2-2-2-privileged-mode.md.

## Summary by Sourcery

Validate and document the sandbox privileged-mode container creation contract, ensuring configuration options that should be ignored are explicitly covered by tests and documentation.

Enhancements:
- Clarify the privileged-mode behaviour so that engine host profiles govern device access and security labelling while sandbox flags are ignored as intended.

Documentation:
- Document that privileged mode ignores FUSE (/dev/fuse) and SELinux label settings, and mark the privileged-mode roadmap task as complete.
- Add an execution plan describing the privileged-mode validation work and its verification steps.

Tests:
- Add unit tests covering privileged-mode edge cases, including ignored FUSE/SELinux settings, config-to-security option mapping, and operation without optional fields.
- Add BDD scenarios confirming at a behavioural level that privileged mode ignores /dev/fuse and SELinux override settings while still using privileged host configuration.

### CODE REVIEW

File: src/engine/connection/create_container/tests.rs [hunk 1]
```diff
@@ -338,3 +338,66 @@ fn create_container_sync_uses_provided_runtime(
...
+    runtime: std::io::Result<tokio::runtime::Runtime>,
+) -> std::io::Result<()> {
+    privileged_create(&runtime?, true, SelinuxLabelMode::DisableForContainer).map(|_| ())
+}
+
+#[rstest]
+fn create_container_privileged_mode_without_optional_fields(
+    runtime: std::io::Result<tokio::runtime::Runtime>,
+) -> std::io::Result<()> {
+    let (opts, body) = privileged_create(&runtime?, true, SelinuxLabelMode::KeepDefault)?;
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/engine/connection/create_container/tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/engine/connection/create_container/tests.rs [hunk 2]
```diff
@@ -338,3 +338,66 @@ fn create_container_sync_uses_provided_runtime(
...
+        .block_on(EngineConnector::create_container_async(&creator, &req))
+        .map_err(|e| io_error(format!("{e}")))?;
+    let body = take_body(&cap).ok_or_else(|| io_error("body"))?;
+    let hc = body.host_config.as_ref().ok_or_else(|| io_error("hc"))?;
+    ensure(hc.privileged == Some(true), "privileged")?;
+    ensure(hc.cap_add.is_none(), "cap_add")?;
+    ensure(hc.devices.is_none(), "devices")?;
+    ensure(hc.security_opt.is_none(), "security_opt")?;
+    Ok((take_options(&cap), body))
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/engine/connection/create_container/tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/engine/connection/create_container/tests/privileged_mode.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+) -> std::io::Result<()> {
+    privileged_create(&runtime?, false, SelinuxLabelMode::KeepDefault).map(|_| ())
+}
+
+#[rstest]
+fn create_container_privileged_mode_ignores_selinux_override(
+    runtime: std::io::Result<tokio::runtime::Runtime>,
+) -> std::io::Result<()> {
+    privileged_create(&runtime?, true, SelinuxLabelMode::DisableForContainer).map(|_| ())
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/engine/connection/create_container/tests/privileged_mode.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
